ipyvuetify/extra/file_input.py

```diff
@@ -148,35 +151,220 @@ def readall(self):
+
+    small_chips = Bool(None, allow_none=True).tag(sync=True)
+
+    solo = Bool(None, allow_none=True).tag(sync=True)
+
+    solo_inverted = Bool(None, allow_none=True).tag(sync=True)
+
+    success = Bool(None, allow_none=True).tag(sync=True)
+
+    success_messages = Union([Unicode(), List(Any())], default_value=None, allow_none=True).tag(sync=True)
+
+    suffix = Unicode(None, allow_none=True).tag(sync=True)
+
+    truncate_length = Union([Float(), Unicode()], default_value=None, allow_none=True).tag(sync=True)
+
+    type = Unicode(None, allow_none=True).tag(sync=True)
+
+    validate_on_blur = Bool(None, allow_none=True).tag(sync=True)
+
+    value = Any(None, allow_none=True).tag(sync=True)
+
+    # VueWidget props
+
+    v_model = List(Dict(), allow_none=True).tag(sync=True)
+
+    style_ = Unicode(None, allow_none=True).tag(sync=True)
+
+    class_ = Unicode(None, allow_none=True).tag(sync=True)
+
+    attributes = Dict(None, allow_none=True).tag(sync=True)
+
+    v_slots = List(Dict()).tag(sync=True, **widget_serialization)
+
+    v_on = Unicode(None, allow_none=True).tag(sync=True)
+
+    # Component-specific props
+
+    template = Unicode(load_template("file_input.vue")).tag(sync=True)
+
+    version = Int(0).tag(sync=True)
+
+    total_progress = Int(0).tag(sync=True)
+
+    progress_indeterminate = Bool(False).tag(sync=True)
+
+    file_info = List(Dict(), allow_none=True)  # Deprecated, use v_model instead
 
     total_progress_inner = 0
     total_size_inner = 0
 
-    def __init__(self, **kwargs):
+    def __init__(self, *args, **kwargs):
         self.chunk_listeners = {}
         self.stats = []
-        super().__init__(**kwargs)
+
+        # Keep default behaviour
+        kwargs.setdefault("multiple", True)
+        kwargs.setdefault("clearable", True)
+
+        attributes = kwargs.setdefault("attributes", {})
+        if "accept" in kwargs:
+            warnings.warn("accept argument is deprecated, use attributes dictionary instead", DeprecationWarning)
+            attributes["accept"] = kwargs["accept"]
+            del kwargs["accept"]
+        if "directory" in kwargs:
+            warnings.warn("directory argument is deprecated, use attributes dictionary instead", DeprecationWarning)
+            attributes["directory"] = kwargs["directory"]
+            del kwargs["directory"]
+        if "show_progress" in kwargs:
+            warnings.warn(
+                "show_progress argument is deprecated, use v_slots to change the progress slot", DeprecationWarning
+            )
+            del kwargs["show_progress"]
+        if "file_info" in kwargs:
+            warnings.warn("file_info cannot be set on FileInput")
+            del kwargs["file_info"]
+        if "v_model" in kwargs:
+            warnings.warn("v_model cannot be set on FileInput")
+            del kwargs["v_model"]
+
+        # Maintain backwards compatibility for the file_info
+        dlink((self, "v_model"), (self, "file_info"))
```

Comment:
I think this is better done on the python side. If we have `file_info` not have sync=True, and observe v_model, and set file_info as a result of that, we don't have to transfer the data twice.

Reply:
I already removed sync tag for `file_info`. It seems that `dlink` does what you described in this case: it updates the target value once on initialization and adds the observer, which modifies the target value on change to the source value.